core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -634,37 +634,40 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
                               responseType: MediaExtension)(implicit transid: TransactionId) = {
     onComplete(queuedActivation) {
       case Success(Right(activation)) =>
-        val result = activation.resultAsJson
+        respondWithActivationIdHeader(activation) {
```

Comment:
I do not think the activation id should be reported in webaction. It leaks information. For normal POSTS it is ok because there’s an explicit subject key. For webactions, there is not.

Reply:
This is to cope with the inconvenience which is described in #3582.
This will give users a better option to retrieve right activation results.

Reply:
A webaction is called by an anonymous subject. The activation id is information specific the subject owning the action.

The webaction can return its own header to identify the action. Which could be the activation id if it chooses - I do not think the system should blanket return the activation id for webactions.